### DISCUSSION TITLE
ScaledJob - Node Scale down not working

### DISCUSSION THREAD

**Discussion by atulthummar** [kedacore/keda#3212] [Q&A / Need Help]
Hi Team 

I have used Service bus - KEDA based scaled job trigger and below are the deployment - It scaled up NODE based on cpu limits as shown below - but once job is finished NODE is not scale down in cluster.
It scale up smoothly but once no more request in Queue we don't want unnecessary node - It should be scale down.

Can you please help us why Node is not scale down 
------------- 

apiVersion: keda.sh/v1alpha1 
kind: ScaledJob
metadata:
  name: usernode-workerapp-scaler
spec:
  jobTargetRef:
    template:
      spec:
        nodeSelector:
          beta.kubernetes.io/os: linux
        containers:
          - name: usernode-workerapp
            image: containerregistry.azurecr.io/v1:latest_gb 
            imagePullPolicy: Always 
            resources:
              limits:
                cpu: 3000m
              requests:
                cpu: 3000m  
            env:
            - name: SERVICEBUS_CONNECTIONSTRING
        imagePullSecrets:
            - name: acr-secret
        restartPolicy: OnFailure
        terminationGracePeriodSeconds: 3600
        tolerations:
        - key: "key"
          operator: "Equal"
          value: "common"
          effect: "NoSchedule"
  successfulJobsHistoryLimit: 0  
  
  triggers:
  - type: azure-servicebus
    metadata:
      queueName: requestqueue
      connectionFromEnv: SERVICEBUS_CONNECTIONSTRING
      namespace: demo
      messageCount: '1'
    authenticationRef:
     name: usernode-workerapp-trigger-auth

![image](https://user-images.githubusercontent.com/84985865/174807812-cde6dd06-5319-4d42-9a54-5e7e51586c7f.png)

**Comment by tomkerkhove** [comment]
I'm not sure if I get your question because KEDA does not scale cluster nodes. Can you elaborate more please?

**Reply by atulthummar** [reply]
I understand KEDA is not maintain Cluster Auto Scaling - I have apply the CPU limitation in POD deployment so that based on each message  required resources and new Node will be scaled up by[ CA](https://docs.microsoft.com/en-us/azure/aks/cluster-autoscaler), However once process is completed by default behavior of CA within 10 min if no pod to schedule it should be terminate node. 

So Can you please  let me know on above deployment is there any impact in Cluster Auto scaling feature ?

**Reply by tomkerkhove** [reply]
CA is not that fast so I think you'll just need to wait some time before it goes away. I recommend asking this question on the official cluster autoscaler repo.

**Reply by zroubalik** [reply]
And you might also want to check Kubernetes Job documentation, what are the ways how to configure the stuff related for resources.

**Reply by atulthummar** [reply]
https://keda.sh/docs/2.4/concepts/scaling-jobs/#details
As I learnt that need to add Scaling Strategy but while applying  the same it throwing below error

error: error validating "D:\\Projects\\Configuration\\Job.yml": error validating data: ValidationError(ScaledJob.spec): unknown field "scalingStrategy" in sh.keda.v1alpha1.ScaledJob.spec; if you choose to ignore these errors, turn validation off with --validate=false

**Reply by atulthummar** [reply]
@tomkerkhove @zroubalik  Any help why Scaling Strategy not apply

**Reply by zroubalik** [reply]
You are probably using some old version of KEDA where that version of CRD doesn't support the new fields. I recommend you to update to latest 2.7.1 and give it a try.

**Reply by atulthummar** [reply]
Thanks its applying after updated version.

  scalingStrategy:
    strategy: "default" 
  triggers:
  - type: azure-servicebus
    metadata:
      queueName: requestqueue
      connectionFromEnv: SERVICEBUS_CONNECTIONSTRING
      namespace: sb-faceapp-demo
      messageCount: '1'
    authenticationRef:
     name: usernode-workerapp-trigger-auth

Still seems node is not scale down after scale up - if you have any suggestion ?
